bundles/org.openhab.core.config.discovery.addon.sddp/src/main/java/org/openhab/core/config/discovery/addon/sddp/SddpAddonFinder.java

```diff
@@ -0,0 +1,172 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.addon.AddonMatchProperty;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.addon.AddonFinder;
+import org.openhab.core.config.discovery.addon.AddonFinderConstants;
+import org.openhab.core.config.discovery.addon.BaseAddonFinder;
+import org.openhab.core.config.discovery.sddp.SddpDevice;
+import org.openhab.core.config.discovery.sddp.SddpDeviceParticipant;
+import org.openhab.core.config.discovery.sddp.SddpDiscoveryService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link SddpAddonFinder} for finding suggested Addons via SDDP.
+ * <p>
+ * It checks the binding's addon.xml 'match-property' elements for the following SDDP properties:
+ * <li>driver</li>
+ * <li>host</li>
+ * <li>ipAddress</li>
+ * <li>macAddress</li>
+ * <li>manufacturer</li>
+ * <li>model</li>
+ * <li>port</li>
+ * <li>primaryProxy</li>
+ * <li>proxies</li>
+ * <li>type</li>
+ * <p>
+ * 
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonFinder.class, name = SddpAddonFinder.SERVICE_NAME)
+public class SddpAddonFinder extends BaseAddonFinder implements SddpDeviceParticipant {
+
+    public static final String SERVICE_TYPE = AddonFinderConstants.SERVICE_TYPE_SDDP;
+    public static final String SERVICE_NAME = AddonFinderConstants.SERVICE_NAME_SDDP;
+
+    private static final String DRIVER = "driver";
+    private static final String HOST = "host";
+    private static final String IP_ADDRESS = "ipAddress";
+    private static final String MAC_ADDRESS = "macAddress";
+    private static final String MANUFACTURER = "manufacturer";
+    private static final String MODEL = "model";
+    private static final String PORT = "port";
+    private static final String PRIMARY_PROXY = "primaryProxy";
+    private static final String PROXIES = "proxies";
+    private static final String TYPE = "type";
+
+    private static final Set<String> SUPPORTED_PROPERTIES = Set.of(DRIVER, HOST, IP_ADDRESS, MAC_ADDRESS, MANUFACTURER,
+            MODEL, PORT, PRIMARY_PROXY, PROXIES, TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(SddpAddonFinder.class);
+    private final Set<SddpDevice> foundDevices = new HashSet<>();
+
+    private @Nullable SddpDiscoveryService sddpDiscoveryService = null;
+
+    @Activate
+    public SddpAddonFinder(
```

Comment:
I doubt this is going to work reliably when there is more than on DiscoveryService. The first one being loaded will get passed in as the reference, and if it is not an SddpDiscoveryService, it will just activate but without a reference. You probably need a separate bind method to set it.

Reply:
That's the problem with the lack of documentation. My feeling is that all conditions must be met before the service can activate. I had MDNS operating at the time I made the test published above, and it did work with the SDDP service being loaded. But as you say, it may be random. I will run some more tests with the logger.warn() to see if I can figure it out.

Reply:
It looks like the @reference is only satisfied when **_both_** the service class name and the service name are satisfied. In other words the service won't be activated until a specific instance of SddpDiscoveryService has been created and loaded.

Reply:
> I doubt this is going to work reliably when there is more than on DiscoveryService. 

@mherwege you were right. The finder @Reference was being satisfied by _any_ DiscoveryService, so sometimes if the first loaded DiscoveryService was an SddpDiscoveryService it worked, and otherwise it did not. I changed it so that the finder @Reference is only satisfied for specific DiscoveryService instances with the `target` attribute matching a @component that has a `property` value of `"protocol=sddp"` .. and it now seems to work predictably..